File: PHP/Mediator/ConcreteMediator.php
<?php


class ConcreteMediator implements Mediator
{
    private $component1;

    private $component2;

    public function __construct(Component1 $c1, Component2 $c2)
    {
        $this->component1 = $c1;
        $this->component1->setMediator($this);
        $this->component2 = $c2;
        $this->component2->setMediator($this);
    }

    public function notify(object $sender, string $event): void
    {
        if ($event == "A") {
            echo "Mediator reacts on A and triggers following operations:<br/>";
            $this->component2->doC();
        }

        if ($event == "D") {
            echo "Mediator reacts on D and triggers following operations:<br/>";
            $this->component1->doB();
            $this->component2->doC();
        }
    }
}

File: Csharp/Behavioral/Mediator/MR1/Mediator/ConcreteMediator.cs
using System;

namespace TheRayCode.Mediator
{
    class ConcreteMediator : Mediator
    {
        private ConcreateColleage1 _concreateColleage1;

        private ConcreateColleage2 _concreateColleage2;

        public ConcreteMediator(ConcreateColleage1 concreateColleage1, ConcreateColleage2 concreateColleage2)
        {
            this._concreateColleage1 = concreateColleage1;
            this._concreateColleage1.SetMediator(this);
            this._concreateColleage2 = concreateColleage2;
            this._concreateColleage2.SetMediator(this);
        } 

        public void Notify(object sender, string ev)
        {
            if (ev == "A")
            {
                Console.WriteLine("Mediator reacts on A and triggers folowing operations:");
                this._concreateColleage2.DoC();
            }
            if (ev == "D")
            {
                Console.WriteLine("Mediator reacts on D and triggers following operations:");
                this._concreateColleage1.DoB();
                this._concreateColleage2.DoC();
            }
        }
    }

}